packages/xo-server/src/xo-mixins/backups-ng/index.js

```diff
@@ -580,24 +579,30 @@ export default class BackupNg {
           (vmsId = extractIdsFromSimplePattern(vmsPattern)) !== undefined
         ) {
           vms = {}
-          const missingVms = []
           vmsId.forEach(id => {
             try {
               vms[id] = app.getObject(id, 'VM')
             } catch (error) {
-              missingVms.push(id)
+              // log failure task in case of missing VM
+              const taskId = logger.notice(
+                `Starting backup of ${id}. (${job.id})`,
+                {
+                  event: 'task.start',
+                  parentId: runJobId,
+                  data: {
+                    type: 'VM',
+                    id,
+                  },
+                }
+              )
+              logger.error(`Backuping ${id} has failed. (${job.id})`, {
+                event: 'task.end',
+                taskId,
+                status: 'failure',
+                result: serializeError(error),
+              })
```

Comment:
I seem to recall that we choose to do a warning instead of failed tasks, but I can't remember why, any idea?

Reply:
I don't remember why, IMHO, it makes sense to log a failure task in case of missing select VM.

Reply:
Agreed.